lib/services/resetpasswordtoken.go

```diff
@@ -0,0 +1,184 @@
+func (u *ResetPasswordTokenV3) GetUser() string {
+	return u.Spec.User
+}
+
+// GetCreated returns Created
+func (u *ResetPasswordTokenV3) GetCreated() time.Time {
+	return u.Spec.Created
+}
+
+// GetURL returns URL
+func (u *ResetPasswordTokenV3) GetURL() string {
+	return u.Spec.URL
+}
+
+// SetURL sets URL
+func (u *ResetPasswordTokenV3) SetURL(url string) {
+	u.Spec.URL = url
+}
+
+// Expiry returns object expiry setting
+func (u *ResetPasswordTokenV3) Expiry() time.Time {
+	return u.Metadata.Expiry()
+}
+
+// SetExpiry sets object expiry
+func (u *ResetPasswordTokenV3) SetExpiry(t time.Time) {
+	u.Metadata.SetExpiry(t)
+}
+
+// SetTTL sets Expires header using current clock
+func (u *ResetPasswordTokenV3) SetTTL(clock clockwork.Clock, ttl time.Duration) {
+	u.Metadata.SetTTL(clock, ttl)
+}
+
+// GetMetadata returns object metadata
+func (u *ResetPasswordTokenV3) GetMetadata() Metadata {
+	return u.Metadata
+}
+
+// GetVersion returns resource version
+func (u *ResetPasswordTokenV3) GetVersion() string {
+	return u.Version
+}
+
+// GetKind returns resource kind
+func (u *ResetPasswordTokenV3) GetKind() string {
+	return u.Kind
+}
+
+// SetName sets the name of the resource
+func (u *ResetPasswordTokenV3) SetName(name string) {
+	u.Metadata.Name = name
+}
+
+// GetResourceID returns resource ID
+func (u *ResetPasswordTokenV3) GetResourceID() int64 {
+	return u.Metadata.ID
+}
+
+// SetResourceID sets resource ID
+func (u *ResetPasswordTokenV3) SetResourceID(id int64) {
+	u.Metadata.ID = id
+}
+
+// GetSubKind returns resource sub kind
+func (u *ResetPasswordTokenV3) GetSubKind() string {
+	return u.SubKind
+}
+
+// SetSubKind sets resource subkind
+func (u *ResetPasswordTokenV3) SetSubKind(s string) {
+	u.SubKind = s
+}
+
+// CheckAndSetDefaults checks and set default values for any missing fields.
+func (u ResetPasswordTokenV3) CheckAndSetDefaults() error {
+	return u.Metadata.CheckAndSetDefaults()
+}
+
+// // String represents a human readable version of the token
+func (u *ResetPasswordTokenV3) String() string {
+	return fmt.Sprintf("ResetPasswordTokenV3(tokenID=%v, user=%v, expires at %v)", u.GetName(), u.Spec.User, u.Expiry())
+}
+
+// NewResetPasswordToken is a convenience wa to create a RemoteCluster resource.
+func NewResetPasswordToken(tokenID string) ResetPasswordTokenV3 {
+	return ResetPasswordTokenV3{
+		Kind:    KindResetPasswordToken,
+		Version: V3,
+		Metadata: Metadata{
+			Name:      tokenID,
+			Namespace: defaults.Namespace,
+		},
+	}
+}
+
+// ResetPasswordTokenSpecV3Template is a template for V3 ResetPasswordToken JSON schema
+const ResetPasswordTokenSpecV3Template = `{
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+		"user": {
+			"type": ["string"]
+		},
+		"created": {
+			"type": ["string"]
+		},
+		"url": {
+			"type": ["string"]
+		}
+  }
+}`
+
+// UnmarshalResetPasswordToken unmarshals ResetPasswordToken
+func UnmarshalResetPasswordToken(bytes []byte) (ResetPasswordToken, error) {
```

Comment:
Do you know why we have a package level lock and interface for all other resources? Wondering if by not doing this here we will run into a problem later.